api/bazel/repositories.bzl

```diff
@@ -287,3 +288,81 @@ go_proto_library(
 )
         """,
     )
+
+    _com_github_grpc_grpc()
+
+def _com_github_grpc_grpc():
```

Comment:
So, what you've done is push a bunch of deps down into `api/bazel`, since we can then have a single API related setup Skylark function. This, to me, seems totally reasonable. the downside is we lose some of the nice machinery for managing repository locations that @jmillikin-stripe setup there. Ideally we also port that back into `api/bazel`, but that might be too much unrelated work for this PR. @jmillikin-stripe do you have any opinions as Envoy Bazel owner?